File: DSAlgorithms/Leetcode/c++/sliding_window/1456-max_vowels_substring.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
       std::unordered_set<char> vowels= {'a', 'e', 'i', 'o', 'u'};
       int left = 0;
       int count = 0;
       int result = 0;
       for(int right = 0; right < s.size(); right++)
       {
           std::unordered_set<char>::iterator find_char = vowels.find(s[right]);
           if(find_char != vowels.end())
                count++;
            if((right - left + 1) > k)
            {
                find_char = vowels.find(s[left]);
                if(find_char != vowels.end())
                    count--;
                left++;
            }
            result = std::max(result, count);
       }
       return result;
    }
};

File: DSAlgorithms/Leetcode/ts/src/sliding_window/1456-max_vowels_substring.ts
const maxVowels = (s: string, k: number): number =>{
    const vowels = new Set<string>(['a', 'e', 'i', 'o', 'u']);
    let left = 0;
    let result = 0;
    let count = 0;
    for(let right = 0; right < s.length; right++)
    {
        if(vowels.has(s[right]))
            count++;
        if((right - left + 1) > k)
        {
            if(vowels.has(s[left]))
                count--;
            left++;
        }
        result = Math.max(result, count);
    }
    return result;

};